### PR TITLE
Improving tar command

### PR DESCRIPTION
**Pull request by pallaud**
In reference to issue #1168, we add:
- -t flag to list contents of a tar archive
- ability to take multiple files/dirs as input

To keep modularity for the future, I broke out some logic into separate
functions (passesFilters, applyToArchive), and added the ability to
apply multiple filters.

I'm not sure about the testing, any suggestions to improve it/make it better organized?

### CODE REVIEW

File: pkg/tar/tar.go [hunk 1]
```diff
@@ -33,58 +35,85 @@ func ExtractDirFilter(tarFile io.Reader, dir string, filter Filter) error {
...
+				f, err := os.Open(path)
+				if err != nil {
+					return err
+				}
+				if _, err := io.Copy(tw, f); err != nil {
+					f.Close()
+					return err
+				}
+				f.Close()
+				// TODO add to a list?
```
Comment (rjoleary) [inline pkg/tar/tar.go]: Can this TODO be made more specific?

File: pkg/tar/tar.go [hunk 2]
```diff
@@ -33,58 +35,85 @@ func ExtractDirFilter(tarFile io.Reader, dir string, filter Filter) error {
...
+	})
+}
+
+// CreateDirs creates a new tar file with all the contents of a directory.
+func CreateDirs(tarFile io.Writer, dirs []string) error {
+	return CreateDirsFilter(tarFile, dirs, nil)
+}
+
+// CreateDirsFilter creates a new tar file with the given filter.
+func CreateDirsFilter(tarFile io.Writer, dirs []string, filters []Filter) error {
```
Comment (rjoleary) [inline pkg/tar/tar.go]: Does `dirs` have to be a list of directories, or does this also support single files?

Comment (pallaud) [reply pkg/tar/tar.go]: It also supports single files - would it be better to rename it `CreateFilesFilter` and have `dirs` be `files` or something like that?

File: pkg/tar/tar.go (hunk 1 above)
Comment (pallaud) [reply pkg/tar/tar.go]: That was a todo I'd made for myself and forgot to remove whoops

File: pkg/tar/tar.go (hunk 2 above)
Comment (rjoleary) [reply pkg/tar/tar.go]: Yes, that would make it easier to understand.

**Comment by rjoleary** [comment]
Looks good to me!

**Pull request merged**